### DISCUSSION TITLE
Is there any support for an iTerm2 "badge" feature?

### DISCUSSION THREAD

**Discussion by rnhurt** [ghostty-org/ghostty#3864] [Feature Requests, Ideas]
iTerm2 allows me to create a dynamic label that appears in the background of a tab.  As a DevOps person, I use this to display which AWS account I'm currently working in.  It's pretty handy to make sure I'm not making updates to the production system.  :grin: 

## Links
- https://iterm2.com/documentation-badges.html

**Comment by gurgeous** [comment]
This seems silly but is actually real handy! I take heavy advantage of this feature and it would be hard to give up. I use several terminals and I mark them with emojis:

<img width="920" alt="image" src="https://github.com/user-attachments/assets/b613c38d-d531-4ddd-a2f7-c9197df4171e" />

**Comment by pluiedev** [comment]
Seems tangentially related to #2509 as we lack any sort of tab/split customization whatsoever. This IMO as a non iTerm2 user seems like way too specific a feature for us to implement. Maybe you could implement this via custom shaders?

**Reply by rnhurt** [reply]
> Seems tangentially related to #2509 as we lack any sort of tab/split customization whatsoever. This IMO as a non iTerm2 user seems like way too specific a feature for us to implement. Maybe you could implement this via custom shaders?

I think I like coloring the tabs better than a badge, at least for my use case.  A bright RED tab helps me remember to not do anything stupid.  :grin: 

The only thing is that the tab color would have to be dynamic.  I change between AWS accounts in a single session.  Changing from my "development" account (green tab) to my "production" account (red tab) should change the tab color.

**Comment by mccolin** [comment]
There are a lot of excellent use cases for the badge:

* Production vs. Non-production distinction (i.e., don't do the bad thing in the red window)
* Identify which project you're working in
* Quickly tell which context is in which split pane (e.g., tail the server log here, the web log here, keep a terminal here, etc.)

In iTerm2 it can also be programmed and connected to `env` variables, so you can set it to change as you switch contexts like going in and out of a virtual env or similar.

Definitely a nice-to-have, but very useful feature.

**Reply by jacob-carlborg** [reply]
For the Production vs. Non-production distinction with iTerm I use a trigger to switch theme to a copy of my theme but with a different background color. This seems like a more general feature, unless already supported.

**Comment by wpueschel** [comment]
I use this feature heavily in iTerm, it's the feature that keeps me from switching to ghostty.
I have it tied into kubie so the badge actually tells me my Kubernetes context and current namespace. Having this in the prompt is way to overloaded, not a fan of big prompts anyway. I usually have only the active directory and time in the prompt.

**Reply by pluiedev** [reply]
Please avoid leaving comments that are to the effect of "I also need this." Clicking the upvote button has the same effect.

**Comment by gurgeous** [comment]
FYI I am giving ghostty another go today and I worked around this a bit using emoji tab titles:

```sh
function tab() { printf "\e]0;%s\a" "$1"; }`
function tab-dev()   { tab "😈"; }
function tab-fire()  { tab "🔥"; }
function tab-ghost() { tab "👻"; }
```

<img width="721" alt="image" src="https://github.com/user-attachments/assets/8f758323-af20-4e6d-9949-7811d3821bf3" />

**Reply by ronalexssen** [reply]
That feels like a good fix. I've run into cases where a command will overwrite the terminal title at which point I'm back to square one.

**Reply by rhodes-b** [reply]
if you are using the tip branch you can just rename the tab to whatever you want in which case no command can override it

**Comment by ronalexssen** [comment]
I filed a Vouch Request for text and emoji badging at https://github.com/ghostty-org/ghostty/discussions/10979

**Reply by gurgeous** [reply]
Yes, please do! I learned a bit of zig recently and I'd be happy to pitch in.

Just to add to the ideas above, I also experimented with using a custom emoji font on macos with `font-codepoint-map` so I can finally get a claude icon into my tab. Sadly, I'm guessing tab titles are native so it doesn't work there, only inside the terminal surface itself. Maybe I need to bite the bullet and use tmux tabs...
